ee/tabby-ui/app/(dashboard)/settings/general/components/network-form.tsx

```diff
@@ -112,15 +115,25 @@ const NetworkForm: React.FC<NetworkFormProps> = ({
 }
 
 export const GeneralNetworkForm = () => {
-  const [{ data }, reexecuteQuery] = useNetworkSetting()
+  const [queryState, reexecuteQuery] = useNetworkSetting({
+    requestPolicy: 'network-only'
+  })
+
+  const { data, fetching, stale } = queryState
+
+  const [shouldInitializeForm] = useDebounceValue(
+    data && !stale && !fetching,
```

Comment:
Because this query is also used elsewhere, it still hits the cache, so a combination of three variables is used to determine the outcome.